### PR TITLE
Rename Doc/README.txt to Doc/README.rst and add formatting

### PR DESCRIPTION
**Pull request by ghost**
I have simply updated the document and its references to README.rst (thanks to whoever formatted it).

### CODE REVIEW

**Comment by ghost** [comment]
CC:  @zware @haypo @methane

File: Doc/README.rst [hunk 1]
```diff
@@ -2,20 +2,20 @@ Python Documentation README
...
-documentation.  You don't need to build them yourself, prebuilt versions are
-available at <https://docs.python.org/dev/download.html>.
+documentation.  You don't need to build them yourself, `prebuilt versions are
+available <https://docs.python.org/dev/download.html>`_.
 
 Documentation on authoring Python documentation, including information about
-both style and markup, is available in the "Documenting Python" chapter of the
-developers guide <https://docs.python.org/devguide/documenting.html>.
+both style and markup, is available in the "`Documenting Python <https://docs.python.org/devguide/documenting.html>`_" chapter of the
+developers guide.
```
Comment (methane) [inline Doc/README.rst]: folding long line is preferred.

**Comment by ghost** [comment]
@methane That line is now folded.

**Review by methane — APPROVED** [review]

**Comment by zware** [comment]
Has anyone tested to make sure the docs build doesn't pick up and build README.rst?  I'm not sure that it would be harmful, but would be unnecessary.

**Comment by methane** [comment]
@zware nice catch!

@rogersachan There are two problem.

* `make check` reports error.  `-i README.rst` should be added to rstlint argument in Makefile.
* `make html` reports warning `README.rst:: WARNING: document isn't included in any toctree`.  Please add `README.rst` to `exclude_patterns` in conf.py.

(I feel markdown is better for README now.  But I was too late.)

**Review by methane — DISMISSED** [review]
Please update Makefile and conf.py

**Comment by ghost** [comment]
@methane I have made the requested changes.

File: Doc/README.rst [hunk 2]
```diff
@@ -108,11 +109,11 @@ see the make targets above).
...
 
-Bugs in the content should be reported to the Python bug tracker at
-https://bugs.python.org.
+Bugs in the content should be reported to the 
+`Python bug tracker <https://bugs.python.org>`_.
 
-Bugs in the toolset should be reported in the Sphinx bug tracker at
-https://www.bitbucket.org/birkenfeld/sphinx/issues/.
+Bugs in the toolset should be reported in the 
+`Sphinx bug tracker <https://www.bitbucket.org/birkenfeld/sphinx/issues/>`_.
```
Comment (vstinner) [inline Doc/README.rst]: This link is outdated, it's now https://github.com/sphinx-doc/sphinx/issues

**Review by vstinner — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
@haypo I have made the requested changes.

**Review by vstinner — APPROVED** [review]

**Comment by vstinner** [comment]
Thanks, I merged your PR.

Right now http://sphinx-doc.org/ link displays "Diese Präsenz ist leider nicht verfügbar." error message in german, whereas http://www.sphinx-doc.org/ redirects to http://www.sphinx-doc.org/en/stable/ which is the expected website. But I hope that the error is only temporary. Otherwise, we can fix that later. I reported this issue on IRC.

**Comment by zooba** [comment]
The Windows installer build picks up this readme file and distributes it as `readme.txt` (see `Tools/msi/exe/exe_files.wxs`, which now needs fixing).

I vote we should keep both so that we can distribute the file without rst formatting.

(As an aside, we also read the /LICENSE file, so if someone wants to come in and "fix" that too then they'll also need to update the installer.)

**Comment by vstinner** [comment]
> The Windows installer build picks up this readme file and distributes it as readme.txt (see Tools/msi/exe/exe_files.wxs, which now needs fixing).

This issue is about Doc/README.txt, but Tools/msi/exe/exe_files.wxs seems to refer to README.txt (root directory).

See PR #2 which renamed README.txt to README.rst. This PR was followed by PR #70, PR #73 and #21.

Is it common to provide README as plain text on Windows?

Would it be possible to provide a README formatted as HTML? It would look nicer than plain text, no? It would also allow to keep new features like images (well, "badges"). For example, on Linux I'm running manually "rst2html README.rst README.html".

**Comment by zooba** [comment]
Ah, I see. This PR just modified the one at the root, which is why it showed up as the last commit for the file.

It doesn't matter whether it's common or not - it's a functional change that should have been on bpo and signed off by the RMs (who are responsible for keeping at least the header in the file up to date). This may just be Ned, depending on where the change was backported to.

**Comment by vstinner** [comment]
I opened http://bugs.python.org/issue29579

**Comment by vstinner** [comment]
I don't think that we need to backport this change to Python 2.7, so I removed the "needs backport to 2.7".

**Pull request merged**